sdl_android/src/main/java/com/smartdevicelink/api/SdlManager.java

```diff
@@ -0,0 +1,516 @@
+    */
+
+	private SdlManager() {}
+
+	private void initialize(){
+		// proxy bridge
+		this.proxyBridge = new ProxyBridge(this);
+		// instantiate managers
+
+		/*
+		this.fileManager = new FileManager(_internalInterface, context);
+		this.lockscreenManager = new LockscreenManager(lockScreenConfig, context, _internalInterface);
+		this.screenManager = new ScreenManager(_internalInterface, this.fileManager);
+		this.permissionManager = new PermissionManager(_internalInterface);
+		this.videoStreamingManager = new VideoStreamingManager(context, _internalInterface);
+		this.audioStreamManager = new AudioStreamManager(_internalInterface);
+		*/
+	}
+
+	private void dispose() {
+		/*
+		this.fileManager.dispose();
+		this.lockscreenManager.dispose();
+		this.audioStreamManager.dispose();
+		this.screenManager.dispose();
+		this.permissionManager.dispose();
+		this.videoStreamingManager.dispose();
+		this.audioStreamManager.dispose();
+		*/
+	}
+
+	// BUILDER
+
+	public static class Builder {
+		SdlManager sdlManager;
+
+		/**
+		 * Main Builder for SDL Manager<br>
+		 *
+		 * The following setters are <strong>REQUIRED:</strong><br>
+		 *
+		 * • setAppId <br>
+		 * • setAppName
+		 */
+		public Builder(){
+			sdlManager = new SdlManager();
+		}
+
+		/**
+		 * Sets the App ID
+		 * @param appId
+		 */
+		public Builder setAppId(@NonNull final String appId){
+			sdlManager.appId = appId;
+			return this;
+		}
+
+		/**
+		 * Sets the Application Name
+		 * @param appName
+		 */
+		public Builder setAppName(@NonNull final String appName){
+			sdlManager.appName = appName;
+			return this;
+		}
+
+		/**
+		 * Sets the Short Application Name
+		 * @param shortAppName
+		 */
+		public Builder setShortAppName(final String shortAppName) {
+			sdlManager.shortAppName = shortAppName;
+			return this;
+		}
+
+		/**
+		 * Sets the Language of the App
+		 * @param hmiLanguage
+		 */
+		public Builder setLanguage(final Language hmiLanguage){
+			sdlManager.hmiLanguage = hmiLanguage;
+			return this;
+		}
+
+		/**
+		 * Sets the TemplateColorScheme for daytime
+		 * @param dayColorScheme
+		 */
+		public Builder setDayColorScheme(final TemplateColorScheme dayColorScheme){
+			sdlManager.dayColorScheme = dayColorScheme;
+			return this;
+		}
+
+		/**
+		 * Sets the TemplateColorScheme for nighttime
+		 * @param nightColorScheme
+		 */
+		public Builder setnightColorScheme(final TemplateColorScheme nightColorScheme){
+			sdlManager.nightColorScheme = nightColorScheme;
+			return this;
+		}
+
+		/*
+		public Builder setLockScreenConfig (final LockScreenConfig lockScreenConfig){
+			sdlManager.lockScreenConfig = lockScreenConfig;
+			return this;
+		}*/
+
+		/**
+		 * Sets the vector of AppHMIType
+		 * @param hmiTypes
+		 */
+		public Builder setAppTypes(final Vector<AppHMIType> hmiTypes){
+
+			if (hmiTypes != null) {
+				sdlManager.hmiTypes = hmiTypes;
+				sdlManager.isMediaApp = hmiTypes.contains(AppHMIType.MEDIA);
+			}else {
```

Comment:
Since this `else` essentially happens in the `build()` block, it really doesn't need to be here. I would set the `hmiTypes` to null. 

```
sdlManager.hmiTypes = hmiTypes;

if (hmiTypes != null) {
    sdlManager.isMediaApp = hmiTypes.contains(AppHMIType.MEDIA);
}
```